### PR TITLE
fix: circular ref results in stackoverflow

### PR DESCRIPTION
**Pull request by haijian-vaadin**
Fixes https://github.com/vaadin/flow/issues/8025

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/connect/BeanValueTypeCheckHelper.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.vaadin.flow.server.connect;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A helper class to break circular nullable checking
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/connect/BeanValueTypeCheckHelper.java]: I would improve the javadoc so as to describe what this class generally does, not what it fixes. Something like: "A helper class representing the cache for storing an already validated Bean values".

File: flow-server/src/main/java/com/vaadin/flow/server/connect/BeanValueTypeCheckHelper.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+class BeanValueTypeCheckHelper {
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/connect/BeanValueTypeCheckHelper.java:12]: Does it being cleared at some point? Do we need a clear method for this cache?

File: flow-server/src/main/java/com/vaadin/flow/server/connect/BeanValueTypeCheckHelper.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    private Map<Type, Set<Object>> cache;
+
+    boolean hasVisited(Object value, Type type) {
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/connect/BeanValueTypeCheckHelper.java:15]: Please add a javadoc for these methods.

File: flow-server/src/main/java/com/vaadin/flow/server/connect/BeanValueTypeCheckHelper.java (hunk 2 above)
Comment (haijian-vaadin) [reply flow-server/src/main/java/com/vaadin/flow/server/connect/BeanValueTypeCheckHelper.java:28]: It relies on Java's garbage collection.

Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/server/connect/BeanValueTypeCheckHelper.java:28]: Ok, it's a disposable cache.

**Pull request merged**